lib/deploygate/xcode/analyze.rb

```diff
@@ -16,12 +16,12 @@ class BundleIdentifierDifferentError < DeployGate::RavenIgnoreException
       # @param [String] build_configuration
       # @param [String] target_scheme
       # @return [DeployGate::Xcode::Analyze]
-      def initialize(workspaces, build_configuration = nil, target_scheme = nil)
+      def initialize(workspaces, build_configuration = nil, target_scheme = nil, xcodeproj_path = nil)
         @workspaces = workspaces
         @build_configuration = build_configuration || DEFAULT_BUILD_CONFIGURATION
         @scheme_workspace = find_scheme_workspace(workspaces)
```

Comment:
`scheme_workspace` should be set from `xcodeproj_path` if supplied, right? 

```ruby
if xcodeproj_path.present?
  @xcodeproj = xcodeproj_path
  @scheme_workspace = "${xcodeproj_path}/${BASE_WORK_DIR_NAME}"
else
  @scheme_workspace = find_scheme_workspace(workspaces)
  @xcodeproj = File.dirname(@scheme_workspace)
end
```

And also, it seems `scheme_workspace` is not used except inferring `xcodeproj` now so this code succeeds. How about fixing `find_scheme_workspace` to `find_xcodeproj` to avoid concerns?

```ruby
def initialize
  ...
  # remove attr_reader :find_scheme_workspace
  @xcodeproj = xcodeproj_path.presence || find_xcodeproj
  ...
end

...

def find_xcodeproj(workspaces)
  // do the same thing in find_scheme_workspace
  scheme_workspace != nil ? File.basename(scheme_workspace) : nil
end
```

Reply:
Thank you. That's right!
I reflected it. https://github.com/DeployGate/deploygate-cli/pull/300/commits/8865f86d1e477964301cf6017a21f61602975e21

Reply:
Sorry. I fixed commit. https://github.com/DeployGate/deploygate-cli/pull/300/commits/d7c77f68c35137e0ecee3a3c8882660f1a4f8292